ops/charm.py

```diff
@@ -9,17 +9,30 @@ class HookEvent(EventBase):
 
 class ActionEvent(EventBase):
 
+    def __init__(self, *args):
+        super().__init__(*args)
+        self._params = None
+
     def defer(self):
         raise RuntimeError('cannot defer action events')
 
     def restore(self, snapshot):
+        self._params = None
         env_action_name = os.environ.get('JUJU_ACTION_NAME')
         event_action_name = self.handle.kind[:-len('_action')].replace('_', '-')
         if event_action_name != env_action_name:
             # This could only happen if the dev manually emits the action, or from a bug.
             raise RuntimeError('action event kind does not match current action')
```

Comment:
I don't understand what restore() is trying to do, as it doesn't actually return anything or mutate any state that actually is preserved.
Is it something we thought we would call but never actually call?

Reply:
The framework calls it because of the nature of how events are emitted (they are persisted and restored once even if they are not deferred). The constructor though is not called for unpickled objects:

See 

https://github.com/canonical/operator/pull/132/files#r375263207
https://github.com/canonical/operator/blob/ed093d5c3fa16454f87639550464376446e1713e/ops/framework.py#L490-L494

Happy to discuss it in more detail.

Reply:
If I see your point correctly, this should address it: https://github.com/canonical/operator/pull/132/commits/7972f8d74424830b8a0bea15ff8791d51d30e475

Reply:
Addressed in #132 and merged into this PR per our discussion.